프로젝트 빌드 실패현상 해결

# 프로젝트 빌드 실패현상 해결

## description
현재 프로젝트를 빌드시 아래와 같은 문구와 함께 빌드에 실패합니다. 
 
```
 ./gradlew build
./gradlew: line 39: cd: "./: No such file or directory
> Task :bootJar FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':bootJar'.
> Main class name has not been configured and it could not be resolved

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
1 actionable task: 1 executed

```

## progress
- [ ] 프로젝트 빌드 실패 현상 해결

--- a/backend/admin/build.gradle
+++ b/backend/admin/build.gradle
@@ -1,2 +1,9 @@
+bootJar {
+    enabled = false
+}
+jar {
+    enabled = true
+}
+
 dependencies {
 }

--- a/backend/build.gradle
+++ b/backend/build.gradle
@@ -1,8 +1,15 @@
-plugins {
-    id 'org.springframework.boot' version '2.3.1.RELEASE'
-    id 'io.spring.dependency-management' version '1.0.9.RELEASE'
-    id 'org.asciidoctor.convert' version '1.5.8'
-    id 'java'
+buildscript {
+    ext {
+        springBootVersion = '2.3.1.RELEASE'
+        dependencyManagementVersion = '1.0.9.RELEASE'
+    }
+    repositories {
+        mavenCentral()
+    }
+    dependencies {
+        classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
+        classpath("io.spring.gradle:dependency-management-plugin:${dependencyManagementVersion}")
+    }
 }
 
 subprojects {
@@ -19,10 +26,6 @@ subprojects {
         mavenCentral()
     }
 
-    ext {
-        set('snippetsDir', file("build/generated-snippets"))
-    }
-
     dependencies {
         implementation 'org.springframework.boot:spring-boot-starter-validation'
         compileOnly 'org.projectlombok:lombok'
@@ -33,13 +36,7 @@ subprojects {
     }
 
     test {
-        outputs.dir snippetsDir
         useJUnitPlatform()
-    }
-
-    asciidoctor {
-        inputs.dir snippetsDir
-        dependsOn test
     }
 
     configurations {

--- a/backend/core/build.gradle
+++ b/backend/core/build.gradle
@@ -1,5 +1,9 @@
-bootJar { enabled = false }
-jar { enabled = true }
+bootJar {
+    enabled = false
+}
+jar {
+    enabled = true
+}
 
 dependencies {
     implementation 'org.springframework.boot:spring-boot-starter-data-jpa'

--- a/backend/web/build.gradle
+++ b/backend/web/build.gradle
@@ -1,3 +1,24 @@
+plugins {
+    id "org.asciidoctor.convert" version "1.5.8"
+}
+
+ext {
+    snippetsDir = file('build/generated-snippets')
+}
+
+asciidoctor {
+    inputs.dir snippetsDir
+    dependsOn test
+}
+
+bootJar {
+    dependsOn asciidoctor
+    from("${asciidoctor.outputDir}/html5") {
+        into 'static/docs'
+    }
+}
+
+
 dependencies {
     implementation project(':core')
     implementation 'net.rakugakibox.spring.boot:logback-access-spring-boot-starter:2.7.1'
@@ -7,4 +28,5 @@ dependencies {
     implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
     testImplementation 'io.rest-assured:rest-assured:3.3.0'
     testImplementation 'org.springframework.restdocs:spring-restdocs-mockmvc'
+    asciidoctor 'org.springframework.restdocs:spring-restdocs-asciidoctor:2.0.1.RELEASE'
 }

